app/src/main/java/com/example/android/cricketscoreboard/StartActivity.java

```diff
@@ -35,14 +39,54 @@ public void onClick(View v) {
         };
 
         startDemoButton.setOnClickListener(clickListener);
+
+        // set onclick listener on start button view
+        startButton = (Button) findViewById(R.id.startBtn);
+
+        startButton.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(v.getContext(), MainActivity.class);
+                intent.putExtra("teamname",teamName.getText().toString());
+                intent.putExtra("batsman1",firstGuyName.getText().toString());
+                intent.putExtra("batsman2", secondGuyName.getText().toString());
+                startActivity(intent);
+                }
+
+        });
+
+
+
+        // set onclick listener on start button view
+        startButton = (Button) findViewById(R.id.startBtn);
```

Comment:
Not sure why this code repeats. You already bound start button and set click listener above